feat(new): Added Azure.AppService.MigrateNodeJs (#2889)
Azure/PSRule.Rules.Azure: docs/en/rules/Azure.AppService.NodeJsVersion.md

    - If the type is `Microsoft.Web/sites/slots/config` and the `name` property is set to `web`, set the `properties.linuxFxVersion` property to `NODE|20-lts`.
- For App Service apps on Windows-based plans:
  - Set the `properties.siteConfig.appSettings.WEBSITE_NODE_DEFAULT_VERSION` property to `~20`.
  - It is also possible to set this for sub-resources by:
    - If the type is `Microsoft.Web/sites/slots`, set the `properties.siteConfig.appSettings.WEBSITE_NODE_DEFAULT_VERSION` property to `~20`.
    - If the type is `Microsoft.Web/sites/config` and the `name` property is set to `appsettings`, set the `properties.WEBSITE_NODE_DEFAULT_VERSION` property to `~20`.
    - If the type is `Microsoft.Web/sites/slots/config` and the `name` property is set to `appsettings`, set the `properties.WEBSITE_NODE_DEFAULT_VERSION` property to `~20`.
    - If the type is `Microsoft.Web/sites/config` and the `name` property is set to `web`, set the `properties.appSettings.WEBSITE_NODE_DEFAULT_VERSION` property to `~20`.
    - If the type is `Microsoft.Web/sites/slots/config` and the `name` property is set to `web`, set the `properties.appSettings.WEBSITE_NODE_DEFAULT_VERSION` property to `~20`.

> For example for Linux:

Reviewer: ```suggestion
For example (Linux app):
```

```json
{
  "type": "Microsoft.Web/sites",
  "apiVersion": "2022-09-01",
  "name": "[parameters('name')]",
  "location": "[parameters('location')]",
  "identity": {
    "type": "SystemAssigned"
  },
  "kind": "web",
  "properties": {
    "serverFarmId": "[resourceId('Microsoft.Web/serverfarms', parameters('planName'))]",
    "httpsOnly": true,
    "siteConfig": {
      "minTlsVersion": "1.2",
      "linuxFxVersion": "NODE|20-lts"
    }
  },
  "dependsOn": [
    "[resourceId('Microsoft.Web/serverfarms', parameters('planName'))]"
  ]
}
```

> For example for Windows:

Reviewer: ```suggestion
For example (Windows app):
```

```json
{
  "type": "Microsoft.Web/sites",
  "apiVersion": "2022-09-01",
  "name": "[parameters('name')]",
  "location": "[parameters('location')]",
  "identity": {
    "type": "SystemAssigned"
  },
  "kind": "web",
  "properties": {
    "serverFarmId": "[resourceId('Microsoft.Web/serverfarms', parameters('planName'))]",
    "httpsOnly": true,

>     "siteConfig": {
>       "minTlsVersion": "1.2",
>       "linuxFxVersion": "~20"
>     }

Reviewer: ```suggestion
    "siteConfig": {
      "minTlsVersion": "1.2",
      "appSettings": [
        {
          "name": "WEBSITE_NODE_DEFAULT_VERSION",
          "value": "~20"
        }
      ]
    }
```

  },
  "dependsOn": [
    "[resourceId('Microsoft.Web/serverfarms', parameters('planName'))]"
  ]
}
```

### Configure with Bicep

To deploy App Services that pass this rule:

> - For App Service apps on Linux-based plans:
>   - Set the `properties.siteConfig.linuxFxVersion` property to `NODE|20-lts`.
>   - It is also possible to set this for sub-resources by:
>     - If the type is `Microsoft.Web/sites/slots`, set the `properties.siteConfig.linuxFxVersion` property to `NODE|20-lts`.
>     - If the type is `Microsoft.Web/sites/config` and the `name` property is set to `web`, set the `properties.linuxFxVersion` property to `NODE|20-lts`.
>     - If the type is `Microsoft.Web/sites/slots/config` and the `name` property is set to `web`, set the `properties.linuxFxVersion` property to `NODE|20-lts`.
> - For App Service apps on Windows-based plans:
>   - Set the `properties.siteConfig.appSettings.WEBSITE_NODE_DEFAULT_VERSION` property to `~20`.
>   - It is also possible to set this for sub-resources by:
>     - If the type is `Microsoft.Web/sites/slots`, set the `properties.siteConfig.appSettings.WEBSITE_NODE_DEFAULT_VERSION` property to `~20`.
>     - If the type is `Microsoft.Web/sites/config` and the `name` property is set to `appsettings`, set the `properties.WEBSITE_NODE_DEFAULT_VERSION` property to `~20`.
>     - If the type is `Microsoft.Web/sites/slots/config` and the `name` property is set to `appsettings`, set the `properties.WEBSITE_NODE_DEFAULT_VERSION` property to `~20`.
>     - If the type is `Microsoft.Web/sites/config` and the `name` property is set to `web`, set the `properties.appSettings.WEBSITE_NODE_DEFAULT_VERSION` property to `~20`.
>     - If the type is `Microsoft.Web/sites/slots/config` and the `name` property is set to `web`, set the `properties.appSettings.WEBSITE_NODE_DEFAULT_VERSION` property to `~20`.

Reviewer: I think we can just simplify this to:

```suggestion
- For Linux-based apps and slots:
  - Set the `properties.siteConfig.linuxFxVersion` property to `NODE|20-lts`.
- For Windows-based apps and slots:
  - Add an app setting within `properties.siteConfig.appSettings` by creating an object with the `name` and `value` properties.
  - Set the `name` property to `WEBSITE_NODE_DEFAULT_VERSION`.
  - Set the `value` property to `~20`.

In addition to setting the `properties.siteConfig` property, you can also use a sub-resource.
```

> For example for Linux:

Reviewer: ```suggestion
For example (Linux app):
```

```bicep
resource linuxWeb 'Microsoft.Web/sites@2022-09-01' = {
  name: name
  location: location
  identity: {
    type: 'SystemAssigned'
  }
  kind: 'web'
  properties: {
    serverFarmId: plan.id
    httpsOnly: true
    siteConfig: {
      minTlsVersion: '1.2'
      linuxFxVersion: 'NODE|20-lts'
    }
  }
}
```

> For example for Windows:

Reviewer: ```suggestion
For example (Windows app):
```

```bicep
resource windowsWeb 'Microsoft.Web/sites@2022-09-01' = {
  name: name
  location: location
  identity: {
    type: 'SystemAssigned'
  }
  kind: 'web'
  properties: {
    serverFarmId: plan.id
    httpsOnly: true

>     siteConfig: {
>       minTlsVersion: '1.2'
>       linuxFxVersion: '~20'
>     }

Reviewer: ```suggestion
    siteConfig: {
      minTlsVersion: '1.2'
      appSettings: [  
        {  
          name: 'WEBSITE_NODE_DEFAULT_VERSION'
          value: '~20
        }  
      ]  
    }
```
